flink-yarn/src/test/java/org/apache/flink/yarn/RegisterApplicationMasterResponseReflectorTest.java

```diff
@@ -88,7 +94,44 @@ public void testGetMethodReflectiveHadoop22() {
 		final RegisterApplicationMasterResponseReflector registerApplicationMasterResponseReflector =
 			new RegisterApplicationMasterResponseReflector(LOG);
 
-		final Method method = registerApplicationMasterResponseReflector.getMethod();
+		final Method method = registerApplicationMasterResponseReflector.getGetContainersFromPreviousAttemptsMethod();
+		assertThat(method, notNullValue());
+	}
+
+	@Test
+	public void testCallsGetSchedulerResourceTypesMethodIfPresent() {
+		final RegisterApplicationMasterResponseReflector registerApplicationMasterResponseReflector =
+			new RegisterApplicationMasterResponseReflector(LOG, HasMethod.class);
+
+		final Optional<Set<String>> schedulerResourceTypeNames =
+			registerApplicationMasterResponseReflector.getSchedulerResourceTypeNamesUnsafe(new HasMethod());
```

Comment:
Instead of calling private methods of the `RegisterApplicationMasterResponseReflector` we could also add an `assumeTrue` statement based on the Hadoop version. Then we can have two tests for Hadoop >= 2.6 and < 2.6.

Reply:
I'm not sure about this. My concern is that, the <2.6 test case might eventually not executed in most cases.

Take `testDoesntCallGetContainersFromPreviousAttemptsMethodIfAbsent` as an example. If we make this test case only executed with Hadoop <2.2, then it's practically not executed. Currently we have Hadoop 2.8 for travis ci test, 2.4 & 2.8 for nightly test, and 2.4 (pom default) for local maven verify unless another version is intentionally specified.

Reply:
Makes sense. I agree that your proposal is better.